app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -531,41 +531,34 @@ private static void recordSuccessfulSyncTime(String username) {
         //this is the temporary implementation of everything past this point
 
         //Wipe storage
-        //TODO: move table instead. Should be straightforward with sandboxed db's
-        wipeStorageForFourTwelveSync();
+        SQLiteDatabase userDb = CommCareApplication.instance().getUserDbHandle();
+        wipeStorageForFourTwelveSync(userDb);
+        userDb.beginTransaction();

```

Comment:
this needs to come before the call to wipeStorage, not after

Reply:
😳